pdns/recursor_cache.cc

```diff
@@ -305,9 +305,11 @@ void MemRecursorCache::replace(time_t now, const DNSName &qname, const QType& qt
   for(const auto i : content) {
     /* Yes, we have altered the d_ttl value by adding time(nullptr) to it
        prior to calling this function, so the TTL actually holds a TTD. */
-    ce.d_ttd=min(maxTTD, static_cast<time_t>(i.d_ttl));   // XXX this does weird things if TTLs differ in the set
-    //    cerr<<"To store: "<<i.d_content->getZoneRepresentation()<<" with ttl/ttd "<<i.d_ttl<<", capped at: "<<maxTTD<<endl;
-    ce.d_records.push_back(i.d_content);
+    if (minTTD == 0 || static_cast<time_t>(i.d_ttl) >= minTTD) {
```

Comment:
I don't think this is the right place to do this check. Unless I'm mistaken, we could end up adding a new entry to the cache without any records, which is wrong.

Reply:
Hmm, thing get complicated because it is not a given all entries have the same TTL.
A scan of the min or max TTL of all records could probaly work... have to think about this after coffee.

Reply:
New code in update to thi PR.